ra/ra_test.go

```diff
@@ -1932,7 +1932,11 @@ func (cr *caaRecorder) IsCAAValid(
 ) (*vaPB.IsCAAValidResponse, error) {
 	cr.Lock()
 	defer cr.Unlock()
-	cr.names[*in.Domain] = true
+	var wildcard bool
+	if in.Wildcard != nil {
+		wildcard = *in.Wildcard
+	}
+	cr.names[*in.Domain] = wildcard
```

Comment:
Does this not break the recorder in the case that the name isn't a wildcard? A lookup in a `map[string]bool` where the key doesn't exist will return `false`.

Reply:
That's why the [corresponding accesses to the `cr.names` map in the test](https://github.com/letsencrypt/boulder/blob/6c6ac31282883b060a7cdfde9b844e540ba9a846/ra/ra_test.go#L2002:L2024) use the two value return form of accessing the map key so that a false value can be discerned from a non-existent key. If you think it would be clearer to make a non-bool type for the map value I could look at that.

Reply:
This ended up changing in bc2a722 - The recorder now holds a `map[string][]caaRecheck` for recording the lookups & I think it's more readable overall.